pkg/agents/team2/config.go

```diff
@@ -77,58 +73,111 @@ func InitTable(numStates int, numActions int) [][]float64 {
 }
 
 func New(baseAgent *infra.Base) (infra.Agent, error) {
-	stateSpace := InitStateSpace(3, 3, 3)
-	actionSpace := InitActionSpace()
-	policies := InitPolicies(27, 3)
-	rTable := InitTable(27, 3)
-	qTable := InitTable(27, 3)
+	actionDim := 6
+	daysAtCriticalDim := baseAgent.HealthInfo().MaxDayCritical + 1
+	stateSpace := InitStateSpace(10, 10, daysAtCriticalDim, 11)
+	actionSpace := InitActionSpace(actionDim)
+	policies := InitPolicies(10*10*daysAtCriticalDim*11, actionDim)
+	averagePolicies := InitTable(10*10*daysAtCriticalDim*11, actionDim)
+	rTable := InitTable(10*10*daysAtCriticalDim*11, actionDim)
+	qTable := InitTable(10*10*daysAtCriticalDim*11, actionDim)
+
 	return &CustomAgent2{
-		Base:          baseAgent,
-		stateSpace:    stateSpace,
-		actionSpace:   actionSpace,
-		policies:      policies,
-		rTable:        rTable,
-		qTable:        qTable,
-		isPlatArrived: false,
+		Base:                  baseAgent,
+		stateSpace:            stateSpace,
+		actionSpace:           actionSpace,
+		policies:              policies,
+		averagePolicies:       averagePolicies,
+		rTable:                rTable,
+		qTable:                qTable,
+		visitCounter:          make([]int, 10*10*daysAtCriticalDim*11),
+		daysAtCriticalCounter: 0,
+		PreviousdayAtCritical: 0,
+		lastAge:               0,
+		neiboughHP:            -1,
+		lastAction:            0,
+		lastFoodTaken:         0,
 	}, nil
 }
 
 func (a *CustomAgent2) Run() {
 	//Communication & Observation
 	//communicate before platform arrives to my floor
+	//ask HP of agent below me
+	a.AskNeighbourHP()
+	a.CheckNeighbourHP()
 
-	/*
-		msg := *messages.NewBaseMessage(a.Floor())
-		a.SendMessage(1, msg)
-	*/
-
-	//Perform the following only when platform arrives
-	//NOTE: should let infra team add a func to see whether the plaftfrom has arrived or not
-	if a.CheckState() != -1 {
-		a.isPlatArrived = true
-	} else {
-		a.isPlatArrived = false
-	}
-	if a.isPlatArrived {
+	//Perform the following only once per day when platform arrives
+	//a.exportQTable()
```

Comment:
Note these functions are commented out. Might be a good idea to delete if they wont be used.